### PR TITLE
feat: improve wordle solving efficiency

### PR DESCRIPTION
**Pull request by iyuangang**
1. Present position located, then filter after positioning.
2. The filtered results become the condition for the next calculation.

### CODE REVIEW

File: src/wordle_solver/solver.py [hunk 1]
```diff
@@ -4,15 +4,17 @@
...
 from selenium.webdriver.common.by import By
 from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support import expected_conditions as ec
 from selenium.webdriver.support.ui import WebDriverWait
-from selenium.webdriver.support import expected_conditions as EC
+
+from definitions import ROOT_DIR
 
 GAME_URL: str = "https://www.powerlanguage.co.uk/wordle/"
+# GAME_URL: str = "https://vue-wordle.netlify.app"
```
Comment (liamganose) [inline src/wordle_solver/solver.py:17]: Backup URL? Did you mean to leave this in?

File: src/wordle_solver/solver.py [hunk 2]
```diff
@@ -22,69 +24,86 @@
...
     for i in range(5):
         if i in word_data["correct"]:
             reg_filter[i] = word_data["correct"][i]
+        elif i in word_data["present_position"]:
+            reg_filter[
+                i] = f"[^{(word_data['absent'] - word_data['present']) | set(word_data['present_position'][i])} ]"
         else:
             reg_filter[i] = f"[^{word_data['absent'] - word_data['present']}]"
     reg_filter = re.compile("".join(reg_filter))
+    logging.info(f"reg_filter: {reg_filter}")
```
Comment (liamganose) [inline src/wordle_solver/solver.py:44]: Consider using `f"{reg_filter=}"` or `f"{reg_filter = }"`

https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

File: src/wordle_solver/solver.py [hunk 3]
```diff
@@ -22,69 +24,86 @@
...
     dbl_letters: set = word_data["absent"] & word_data["present"]
+
     def _count_dbls(word):
         for letter in dbl_letters:
             if word.count(letter) > 1:
                 return False
         return True
+
     words = list(filter(_count_dbls, words))
-
```
Comment (liamganose) [inline src/wordle_solver/solver.py:52]: Good catch on all of these, vim usually helps spot these but was working on VS code on another laptop without my pre-commit linter :(

File: src/wordle_solver/solver.py [hunk 4]
```diff
@@ -22,69 +24,86 @@
...
             reg_filter[i] = f"[^{word_data['absent'] - word_data['present']}]"
     reg_filter = re.compile("".join(reg_filter))
+    logging.info(f"reg_filter: {reg_filter}")
     words = list(filter(reg_filter.match, words))
+    logging.info(f"After filter left {len(words)} words: {words}")
 
     # filter on present letters
-    for letter in word_data["present"]:
-        words = list(filter(lambda word: letter in word, words))
+    for i in word_data["present"]:
```
Comment (liamganose) [inline src/wordle_solver/solver.py:49]: I usually like keeping i for incremental integers (range(x)) and using descriptive variable names when taking elements. Nothing specified in pep8 so personal preference I suppose.

**Review by liamganose — APPROVED** [review]
Thanks for finding this and improving. The logic looks great and all the changes are really good, left a couple of comments but otherwise lgtm.

File: src/wordle_solver/solver.py (hunk 1 above)
Comment (iyuangang) [reply src/wordle_solver/solver.py:17]: My bad, it's a backup link.

**Pull request merged**